### ISSUE TITLE
KeyError from 422 response lacking an error object

### ISSUE THREAD

**Issue by Mariatta** [brettcannon/gidgethub#49]
Seeing this in miss-islington's logs. Thought I'll report it :)

<img width="811" alt="screen shot 2018-03-05 at 3 28 06 pm" src="https://user-images.githubusercontent.com/5844587/37005794-43b908e8-208a-11e8-901c-aac3fb9e26be.png">

```
Mar 05 10:45:08 miss-islington app/web.1: Traceback (most recent call last): 
Mar 05 10:45:08 miss-islington app/web.1:   File "/app/miss_islington/__main__.py", line 42, in main 
Mar 05 10:45:08 miss-islington app/web.1:     await router.dispatch(event, gh) 
Mar 05 10:45:08 miss-islington app/web.1:   File "/app/.heroku/python/lib/python3.6/site-packages/gidgethub/routing.py", line 80, in dispatch 
Mar 05 10:45:08 miss-islington app/web.1:     await callback(event, *args, **kwargs) 
Mar 05 10:45:08 miss-islington app/web.1:   File "/app/miss_islington/delete_branch.py", line 25, in delete_branch 
Mar 05 10:45:08 miss-islington app/web.1:     await gh.delete(url) 
Mar 05 10:45:08 miss-islington app/web.1:   File "/app/.heroku/python/lib/python3.6/site-packages/gidgethub/abc.py", line 110, in delete 
Mar 05 10:45:08 miss-islington app/web.1:     await self._make_request("DELETE", url, url_vars, b"", accept) 
Mar 05 10:45:08 miss-islington app/web.1:   File "/app/.heroku/python/lib/python3.6/site-packages/gidgethub/abc.py", line 66, in _make_request 
Mar 05 10:45:08 miss-islington app/web.1:     data, self.rate_limit, more = sansio.decipher_response(*response) 
Mar 05 10:45:08 miss-islington app/web.1:   File "/app/.heroku/python/lib/python3.6/site-packages/gidgethub/sansio.py", line 288, in decipher_response 
Mar 05 10:45:08 miss-islington app/web.1:     errors = data["errors"] 
Mar 05 10:45:08 miss-islington app/web.1: KeyError: 'errors' 
```

Using gidgethub 2.4.1.

**Comment by brettcannon** [comment]
So https://developer.github.com/v3/#client-errors suggests that shouldn't be an issue when GitHub returns a 422 on a request. I guess `data.get('errors', None)` along with some customization of the error message to make sense when there is no error object?

**Comment by Mariatta** [comment]
> data.get('errors', None) along with some customization of the error message to make sense when there is no error object

That sounds good to me. Thanks :)

**Comment by Mariatta** [comment]
To provide more context, the error happened in miss-islington because it tried to delete a branch that was already deleted.

I've tested this again just now, it seems that it was GitHub not following their own documentation, and it did not provide the "errors" object.

The response I got is the following:
status code 422: 
```
{"message":"Reference does not exist",
"documentation_url":"https://developer.github.com/v3/git/refs/#delete-a-reference"}
```
If it's ok, I can try propose a PR for this situation, and will use "message": "Reference does not exist" as the error message.
